src/sentry/integrations/repository/generic.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
```

Comment:
basically a copy of Issue/Metric repositories, but generic and uses `composite_key` and `group_id`